src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -88,16 +85,43 @@ struct AreaStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::
      */
     AreaStructureWrapper & operator=(const AreaStructureWrapper & aOther)
     {
+        areaID = aOther.areaID;
+        mapID  = aOther.mapID;
+
         if (aOther.areaDesc.locationInfo.IsNull())
         {
-            Set(aOther.areaID, aOther.mapID, CharSpan(), NullOptional, NullOptional, aOther.areaDesc.landmarkTag,
-                aOther.areaDesc.positionTag, aOther.areaDesc.surfaceTag);
+            areaDesc.locationInfo.SetNull();
         }
         else
         {
-            Set(aOther.areaID, aOther.mapID, aOther.areaDesc.locationInfo.Value().locationName,
-                aOther.areaDesc.locationInfo.Value().floorNumber, aOther.areaDesc.locationInfo.Value().areaType,
-                aOther.areaDesc.landmarkTag, aOther.areaDesc.positionTag, aOther.areaDesc.surfaceTag);
+            areaDesc.locationInfo.SetNonNull();
+
+            // deep copy the name.
+            auto sizeToCopy = std::min(sizeof(mLocationNameBuffer), aOther.areaDesc.locationInfo.Value().locationName.size());
```

Comment:
Recommend using CopyCharSpanToMutableCharSpan here - it already safely implements the sizing.

Reply:
@cecille Tracked using https://github.com/project-chip/connectedhomeip/issues/34858. We'll address it in a subsequent PR (Will has several more in the pipeline).